src/Components/CodeLensHelpers.fs

```diff
@@ -2,26 +2,27 @@ namespace Ionide.VSCode.FSharp
 
 open System
 open Fable.Core.JsInterop
-open Fable.Import.vscode
+open Fable.Import.VSCode
+open Fable.Import.VSCode.Vscode
 open global.Node
 
 module node = Node.Api
 
 module CodeLensHelpers =
 
     let showReferences (args: string) (args2: obj) (args3: obj[])   =
-        let uri = Uri.parse args
-        let pos = Position(!!args2?Line, !!args2?Character)
+        let uri = vscode.Uri.parse args
+        let pos = vscode.Position.Create(!!args2?Line, !!args2?Character)
         let locs =
             args3
             |> Seq.map (fun f ->
-                let uri = Uri.parse !!f?Uri
-                let range = Range(!!f?Range?Start?Line, !!f?Range?Start?Character, !!f?Range?End?Line, !!f?Range?End?Character)
-                Location(uri, !^ range)
+                let uri = vscode.Uri.parse !!f?Uri
+                let range = vscode.Range.Create(!!f?Range?Start?Line, !!f?Range?Start?Character, !!f?Range?End?Line, !!f?Range?End?Character)
+                vscode.Location.Create(uri, !^ range)
             )
             |> ResizeArray
-        commands.executeCommand("editor.action.showReferences", uri, pos, locs )
+        commands.executeCommand("editor.action.showReferences", ResizeArray [Some (box uri); Some (box pos); Some (box locs);])
 
     let activate (context : ExtensionContext) =
 
-        commands.registerCommand("fsharp.showReferences", showReferences |> objfy4 ) |> context.subscriptions.Add
\ No newline at end of file
+        commands.registerCommand("fsharp.showReferences", showReferences |> objfy4 ) |> box |> unbox context.subscriptions.Add
```

Comment:
you'll see this pattern _a lot_ because the types from `registerCommand` (IDisposable) don't line up with the types of `context.subscriptions.Add` (aka `ResizeArray<ExtensionContextSubscriptions>.Add`, aka ExtensionContextSubscriptions -> unit`).

Disposable and ExtensionContextSubscriptions have the same shapes:

```fsharp
    type [<AllowNullLiteral>] ExtensionContextSubscriptions =
        abstract dispose: unit -> obj option
```

```fsharp
    /// Represents a type which can release resources, such
    /// as event listening or a timer.
    type [<AllowNullLiteral>] Disposable =
        /// Dispose this object.
        abstract dispose: unit -> obj option
```

but we need some way to coerce between them (and other semantically-similar types) in a safe way.

Reply:
I added a small SRTP member to the ExtensionContext type that cleared up a lot of this noise